docs/content/014-guides/022-source-types.mdx

```diff
@@ -0,0 +1,449 @@
+
+In that case, you can configure your `source` to teach Nexus about your convention using regexes:
+
+```ts
+//db.ts
+
+export class UserModel {
+// ^-------------^---------- Export your source type (required)
+//               |---------- Name of your source type which will be extracted by Nexus
+  firstName: string
+  lastName: string
+}
+
+export class PostModel {
+// ^-------------^---------- Export your source type (required)
+//               |---------- Name of your source type which will be extracted by Nexus
+// ...
+}
+
+export class CommentModel {
+// ^-------------^---------- Export your source type (required)
+//               |---------- Name of your source type which will be extracted by Nexus
+// ...
+}
+
+// schema.ts
+import { makeSchema, objectType, queryType } from '@nexus/schema'
+import * as path from 'path'
+
+const User = objectType({
+  name: 'User',
+  //      ^---------------------- Name of your GraphQL Object type which will be matched against the name of your source types
+  definition(t) {
+    t.string('fullName', {
+      resolve(user) {
+        //      ^------------------------ as: UserModel
+        return user.firstName + ' ' + user.lastName
+      }
+    })
+  }
+})
+
+const Query = queryType({
+  definition(t) {
+    t.list.field('users', {
+      type: User,
+      resolve(_parent, args, ctx) {
+        //                       ^------- return as: UserModel[]
+        // ...
+      }
+    })
+  }
+})
+
+const schema = makeSchema({
+  types: [User, Query],
+  typegenAutoConfig: {
+    sources: [{
+      source: path.join(__dirname, 'db.ts'),
+      //       ^----------------------------- Path to the file containing your source types
+      alias: 'db',
+      //      ^------------------------------ Name used as alias for Nexus to import your source types. eg: import * as <name> from <path>
+      typeMatch(type, defaultRegex) {
+      //         ^--------^---------------- GraphQL Named type to match against your source types
+      //                  |---------------- Default regex used to match your GraphQL types against your source types
+        return new RegExp(`(?:interface|type|class|enum)\\s+(${type.name}Model)\\W`, 'g')
+      //            ^---------------------- New regex which will match `UserModel` with `User`, etc..
+      },
+    }]
+  }
+})
+```
+
+You can also use multiple regexes to map your source types by returning an array of regexes:
+
+```ts
+const schema = makeSchema({
+  typegenAutoConfig: {
+    sources: [{
+      source: path.join(__dirname, 'db.ts'),
+      alias: 'db',
+      typeMatch(type, defaultRegex) {
+        return [
+          defaultRegex,
+          //    ^---------------- Pass in the default Regex to keep the default behavior
+          new RegExp(`(?:interface|type|class|enum)\\s+(${type.name}Model)\\W`, 'g'),
+          // new RegExp(...), -> Add more regexes if you have even more convention
+        ]
+      },
+    }]
+  }
+})
+```
+
```

Comment:
What happens when the regex does not match (or none of the multiple given do)? Error? Any?